searx/webapp.py

```diff
@@ -247,37 +254,7 @@ def _get_browser_or_settings_language(req, lang_list):
         locale = match_language(lang, lang_list, fallback=None)
         if locale is not None:
             return locale
-    return settings['search']['default_lang'] or 'en'
-
-
-@babel.localeselector
-def get_locale():
-    if 'locale' in request.form\
-       and request.form['locale'] in LOCALE_NAMES:
-        # use locale from the form
-        locale = request.form['locale']
-        locale_source = 'form'
-    elif request.preferences.get_value('locale') != '':
```

Comment:
See pre_request:
https://github.com/searxng/searxng/blob/a925254c3d19c6a3badf43469816fc1b497e2990/searx/webapp.py#L547-L548

If `not request.preferences.get_value('locale')` then the value comes from `get_locale()` which return `_get_browser_or_settings_language(request, UI_LOCALE_CODES) in this case.